### PR TITLE
email improvements

### LINKED ISSUE #2175: Add Auto-Submitted header to outgoing emails
https://tools.ietf.org/html/rfc3834

Header "Auto-Submitted: autogenerated" should be present in all outgoing messages

### LINKED ISSUE #2190: Public remark email is not sent to the applicant, only handlers
- [x] public remark is sent to applicant
- consider whether to send it to the members also

### PR DESCRIPTION
**Pull request by opqdonut**
- notify applicant & members about public remarks: fixes #2190
- improve working of remark email a bit
- add Auto-Submitted header: fixes #2175

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue

## Documentation
- [x] update changelog if necessary

## Testing
- [x] valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [x] no critical TODOs left to implement

### CODE REVIEW

File: src/clj/rems/email/template.clj [hunk 1]
```diff
@@ -150,7 +150,9 @@
                         :t.email.reviewed/message))
 
 (defmethod event-to-emails :application.event/remarked [event application]
-  (emails-to-recipients (handlers application)
+  (emails-to-recipients (concat (handlers application)
+                                (when (:application/public event)
+                                  (application-util/applicant-and-members application)))
```
Comment (Macroz) [inline src/clj/rems/email/template.clj:155]: I think we sort of agreed on the channel that perhaps this sort of communication would go only to the applicant and the most important one i.e. approval or reject to all?

File: test/clj/rems/email/test_template.clj [hunk 2]
```diff
@@ -187,8 +189,18 @@
...
-           (email-to "assistant" mails)))))
+            :body "Dear Amber Assistant,\n\nRandom Remarker has commented on the application 2001/3, \"Application title\", submitted by Alice Applicant.\n\nYou can review the application at http://example.com/application/7"}
+           (email-to "assistant" mails))))
+  (let [mails (emails base-events {:application/id 7
+                                   :event/type :application.event/remarked
+                                   :event/actor "remarker"
+                                   :application/public true
+                                   :application/comment "remark!"})]
+    (is (= #{"assistant" "handler" "applicant"} (email-recipients mails)))
+    (is (= {:to-user "applicant"
```
Comment (Macroz) [inline test/clj/rems/email/test_template.clj:200]: Do we have some test that checks member mails too?

File: test/clj/rems/email/test_core.clj [hunk 3]
```diff
@@ -76,3 +80,25 @@
               "should skip the message quietly"))
         (is (nil? @message-atom))
         (reset! message-atom nil)))))
+
+(deftest test-send-email!-mock-smtp
+  ;; run one case with an smtp server to make sure the Auto-Submitted header is set
+  (let [port 3025
+        server (GreenMail. (ServerSetup. port nil ServerSetup/PROTOCOL_SMTP))]
```
Comment (Macroz) [inline test/clj/rems/email/test_core.clj:87]: Nice testing emails!

File: src/clj/rems/email/template.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/email/template.clj:155]: Yeah, I used `applicant-and-members` here for consistency with the other emails. Should we remove it for every email?

File: test/clj/rems/email/test_template.clj (hunk 2 above)
Comment (opqdonut) [reply test/clj/rems/email/test_template.clj:200]: the bodies are known to be the same so the tests check only one

File: test/clj/rems/email/test_core.clj (hunk 3 above)
Comment (opqdonut) [reply test/clj/rems/email/test_core.clj:87]: Yeah this GreenMail was a nice find, I had never heard of it before.

File: test/clj/rems/email/test_template.clj (hunk 2 above)
Comment (opqdonut) [reply test/clj/rems/email/test_template.clj:200]: Oh right, this application doesn't have any members.

File: src/clj/rems/email/template.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/email/template.clj]: After discussion on Flowdock, removed it from only this case.

**Pull request merged**